debezium-core/src/main/java/io/debezium/relational/TableId.java

```diff
@@ -37,6 +37,12 @@ public static TableId parse(String str) {
     public static TableId parse(String str, boolean useCatalogBeforeSchema) {
         String[] parts = str.split("[\\" + '.' + "]");
         if ( parts.length < 0 ) return null;
+        for (int i = 0; i < parts.length; i++) {
+            String part = parts[i];
+            if ( part.startsWith("\"") && part.endsWith("\"") ) {
+                parts[i] = part.substring(1, part.length()-1);
+            }
+        }
```

Comment:
It seems to me that the transformation methods will be the same for all databases - just quoting chars will be different. How about providing generic lambda functions tha will take as a parameter start and end char and then make a specialized functions per database by currying the generic functions with database specific chars?